internal/uvm/start.go

```diff
@@ -1,6 +1,63 @@
 package uvm
 
+import (
+	"encoding/json"
+	"io"
+	"io/ioutil"
+	"net"
+	"syscall"
+
+	"github.com/sirupsen/logrus"
+)
+
+const _ERROR_CONNECTION_ABORTED syscall.Errno = 1236
+
+func forwardGcsLogs(l net.Listener) {
+	c, err := l.Accept()
+	if err != nil {
+		logrus.Error("accepting log socket: ", err)
+		return
+	}
+	j := json.NewDecoder(c)
+	logger := logrus.StandardLogger()
+	for {
+		e := logrus.Entry{Logger: logger}
+		err = j.Decode(&e.Data)
+		if err == io.EOF || err == _ERROR_CONNECTION_ABORTED {
+			break
+		}
+		if err != nil {
+			logrus.Error("gcs log read: ", err)
+			rest, _ := ioutil.ReadAll(io.MultiReader(j.Buffered(), c))
```

Comment:
I don't grasp this logic. This will return the raw string that failed to decode in `j.Buffered()`. What will be read from `c`? If the gcs doesn't stop writing does the `Read(c)` ever return before it allocates `rest` as an entire log sequence?

Reply:
This is basically for the scenario where gcs panics and writes the panic stack to stderr. In that case, I think we want to read the rest of the message as a single log entry. If there's a GCS bug and it writes random non-JSON stuff to stderr, then yes this will hang until the VM is shut down.

Reply:
Ok. As long as we all agree this is 100% unsafe. I'm good with it